src/format/formatting.rs

```diff
@@ -46,28 +46,17 @@ pub struct DelayedFormat<I> {
 }
 
 #[cfg(feature = "alloc")]
-impl<'a, I: Iterator<Item = B> + Clone, B: Borrow<Item<'a>>> DelayedFormat<I> {
+impl<'a, I, B> DelayedFormat<I>
+where
+    I: Iterator<Item = B> + Clone,
+    B: Borrow<Item<'a>>,
+{
```

Comment:
As a note: `DelayedFormat::{new, new_with_locale}` are in an impl block without a generic offset parameter. Because they don't take an offset as argument it would fail during type inference otherwise.

`new_with_offset` is in an impl block with the new generic offset parameter.